src/circuit.rs

```diff
@@ -2,61 +2,771 @@
 
 use std::mem;
 
+use group::{Curve, GroupEncoding};
 use halo2::{
     circuit::{Layouter, SimpleFloorPlanner},
-    plonk,
+    plonk::{self, Advice, Column, Expression, Instance as InstanceColumn, Selector},
     poly::Rotation,
     transcript::{Blake2bRead, Blake2bWrite},
 };
-use pasta_curves::{pallas, vesta};
+use pasta_curves::{
+    arithmetic::{CurveAffine, FieldExt},
+    pallas, vesta,
+};
 
 use crate::{
-    note::{nullifier::Nullifier, ExtractedNoteCommitment},
-    primitives::redpallas::{SpendAuth, VerificationKey},
+    constants::{
+        load::{NullifierK, OrchardFixedBasesFull, ValueCommitV},
+        MERKLE_DEPTH_ORCHARD,
+    },
+    keys::{
+        CommitIvkRandomness, DiversifiedTransmissionKey, NullifierDerivingKey, SpendValidatingKey,
+    },
+    note::{
+        commitment::{NoteCommitTrapdoor, NoteCommitment},
+        nullifier::Nullifier,
+        ExtractedNoteCommitment,
+    },
+    primitives::{
+        poseidon::{self, ConstantLength},
+        redpallas::{SpendAuth, VerificationKey},
+    },
+    spec::NonIdentityPallasPoint,
     tree::Anchor,
-    value::ValueCommitment,
+    value::{NoteValue, ValueCommitTrapdoor, ValueCommitment},
+};
+use gadget::{
+    ecc::{
+        chip::{EccChip, EccConfig},
+        FixedPoint, FixedPointBaseField, FixedPointShort, Point,
+    },
+    poseidon::{
+        Hash as PoseidonHash, Pow5T3Chip as PoseidonChip, Pow5T3Config as PoseidonConfig,
+        StateWord, Word,
+    },
+    sinsemilla::{
+        chip::{SinsemillaChip, SinsemillaConfig, SinsemillaHashDomains},
+        commit_ivk::CommitIvkConfig,
+        merkle::{
+            chip::{MerkleChip, MerkleConfig},
+            MerklePath,
+        },
+        note_commit::NoteCommitConfig,
+    },
+    utilities::{
+        copy,
+        enable_flag::{EnableFlagChip, EnableFlagConfig},
```

Comment:
Does not compile.

Reply:
I've fixed this: https://github.com/zcash/orchard/pull/101#issuecomment-882814189